apt/private/resolve.bzl

```diff
@@ -149,16 +105,22 @@ def _deb_resolve_impl(rctx):
     lockf = internal_resolve(rctx, rctx.attr.yq_toolchain_prefix, rctx.attr.manifest, rctx.attr.resolve_transitive)
     lockf.write("lock.json")
 
-    locklabel = rctx.attr.manifest.relative(rctx.attr.manifest.name.replace(".yaml", ".lock.json"))
+    lock_filename = rctx.attr.manifest.name.replace(".yaml", ".lock.json")
+    lock_label = rctx.attr.manifest.relative(lock_filename)
+    workspace_relative_path = "{}{}".format(
+        ("%s/" % lock_label.package) if lock_label.package else "",
+        lock_label.name,
+    )
+
     rctx.file(
         "copy.sh",
-        _COPY_SH_TMPL.format(
+        rctx.read(Label("//apt/private:copy.sh.tmpl")).format(
```

Comment:
This lead to repository restarts, which is not a good thing. can you revert this back to be a static string here?

Reply:
If we want this to be subject to shellcheck, then you should create a private attribute to this repository rule and use this label as the default value to prevent repository rule restarts.

Reply:
Oh wow! I don't even understand what you are saying 😅 I'm quite a Bazel n00b so I don't know what a "repo restart" is. You mean the `Label` in there  causes "cache problems" / somehow makes the build "not immutable" or something and it causes a rebuild or some other side-effect!? :-?  I'd love to learn more about this!

Re. creating a private attribute, would something like this work?
```starlark
def _deb_resolve_impl(rctx):
    lockf = internal_resolve(rctx, rctx.attr.yq_toolchain_prefix, rctx.attr.manifest, rctx.attr.resolve_transitive)
    lockf.write("lock.json")

    lock_filename = rctx.attr.manifest.name.replace(".yaml", ".lock.json")
    lock_label = rctx.attr.manifest.relative(lock_filename)
    workspace_relative_path = "{}{}".format(
        ("%s/" % lock_label.package) if lock_label.package else "",
        lock_label.name,
    )

    rctx.file(
        "copy.sh",
        rctx.read(rctx._copy_sh_tmpl).format(
            # NOTE: the split("~") is needed when we run bazel from another
            # directory, e.g. when running e2e tests we change dir to e2e/smoke
            # and then rctx.name is 'rules_distroless~~apt~bullseye'
            repo_name = rctx.name.split("~")[-1].replace("_resolve", ""),
            lock_label = lock_label,
            workspace_relative_path = workspace_relative_path,
        ),
        executable = True,
    )

    rctx.file("BUILD.bazel", _BUILD_TMPL)

deb_resolve = repository_rule(
    implementation = _deb_resolve_impl,
    attrs = {
        "manifest": attr.label(),
        "resolve_transitive": attr.bool(default = True),
        "yq_toolchain_prefix": attr.string(default = "yq"),
        "_copy_sh_tmpl": attr.label(
            defaul="//apt/private:copy.sh.tmpl",
            doc = "INTERNAL, DO NOT USE - " +
                  "private attribute label to prevent repo restart",
        ),
    },
)
```
Thanks!

Reply:
Also, re. shellcheck, it won't work straight on the template "obviously" because of the templated variables... it's just much easier to do the manual work of copying the template, changing those variables and running `shellcheck`. Plus IMO it makes the Bazel code / the file much nicer and easier to read 😄  But I'm also happy to put the script back into a string if it's easier / prevents other issues that I don't fully grasp 😅

Reply:
Ah, I've asked chatGPT 😆😆

> By using `rctx.read(Label("//apt/private:copy.sh.tmpl"))`, Bazel will re-read the `copy.sh.tmpl` file every time the repository rule runs, which can trigger repository restarts. Repository restarts happen when Bazel notices a change in an external repository, which can lead to performance issues because Bazel will re-run the repository rule and reload dependencies, **even if the file hasn't actually changed**.
>
> By using a private attribute with a default value (like `Label("//apt/private:copy.sh.tmpl")`), you give the rule an explicit, static dependency on this file. This prevents Bazel from thinking the file might change during rule execution, avoiding unnecessary repository restarts.

Let me know if this is correct! :)  I wouldn't have thought that it can make a difference whether the `Label` is an attribute or not 😄  TIL!